app/modules/transcribe/subjects.factory.js

```diff
@@ -39,7 +39,8 @@ function SubjectsFactory($q, AnnotationsFactory, localStorageService, zooAPI, zo
             _updateStorage();
         }
         if (!_queue.length) {
-            return _populateQueue()
+            return  _getSubjectSetId()
+                .then(_populateQueue)
```

Comment:
I'm probably missing angular magic but does this use the same subjectSet state param that has been set via the genre UI selection or hte initial random workflow selection?

Reply:
It is. I didn't know I could do that when chaining promises, but Rog showed me the way.

Reply:
TIL :)